packages/storybook/src/components/tokensTable/TokensTable.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Table } from '@storybook/components';
+import './TokensTable.scss';
+
+/**
+ * Interface defining the structure of a token table's properties
+ */
+export interface TokenTableProps {
+  /** The category name for the tokens being displayed */
+  category: string;
+  /** Array of tokens to display in the table */
+  tokens: Array<{
+    /** The token name/identifier */
+    name: string;
+    /** Visual preview component for the token */
+    preview: React.ReactNode;
+    /** The token's value */
+    value: string;
+  }>;
+}
+
+/**
+ * TokensTable component displays a table of design tokens with their names, values, and previews.
+ * It includes functionality to copy token names to clipboard with visual feedback.
+ *
+ * @component
+ * @example
+ * ```tsx
+ * <TokensTable
+ *   category="Colors"
+ *   tokens={[
+ *     { name: '--ods-color-primary-500', value: '#0050D7', preview: <ColorPreview /> }
+ *   ]}
+ * />
+ * ```
+ */
+export const TokensTable: React.FC<TokenTableProps> = ({ category, tokens }) => {
+  const [copiedToken, setCopiedToken] = useState<string | null>(null);
+
+  const copyToClipboard = (text: string): void => {
+    navigator.clipboard.writeText(text);
+    setCopiedToken(text);
+
+    setTimeout(() => {
+      setCopiedToken(null);
+    }, 2000);
+  };
+
+  return (
+    <Table>
+      <thead>
+        <tr>
+          <th>Token</th>
+          <th>Value</th>
+          <th>Preview</th>
```

Comment:
would probably be better to add a dedicated copy button with tooltip (like clipboard or code)